service/order.go

```diff
@@ -359,15 +400,114 @@ func (s *OrderService) Delete(ctx context.Context, in *pb.OrderSpecificRequest)
 // Lock locks the given Order if the Order is created by this node, broadcasts the lock to other nodes on the channel.
 func (s *OrderService) Lock(ctx context.Context, in *pb.OrderSpecificRequest) (*pb.Empty, error) {
 
-	// TODO: Add Order locking logic
+	orderInBytes, err := s.Storage.Get(getOrderStorageKey(in.GetChannelID(), in.GetOrderID()))
+	if !errors.IsEmpty(err) {
+		return nil, errors.E(errors.Op("Get order in Lock"), err)
+	}
+
+	order := &pb.Order{}
+	err = proto.Unmarshal(orderInBytes, order)
+	if !errors.IsEmpty(err) {
+		return &pb.Empty{}, errors.E(errors.Op("Unmarshal order proto in Lock"), err)
```

Comment:
In these cases it's unnecessary to return &pb.Empty{} instead of just `nil`. Sorry if I haven't spotted this earlier in similar cases.

Reply:
It's fine